components/ActivityPage.tsx

```diff
@@ -1,10 +1,236 @@
-import React from 'react';
-import { View, Text } from 'react-native';
+import React, { useState, ChangeEvent } from 'react';
+import RNDateTimePicker from '@react-native-community/datetimepicker';
+import { daysOfTheWeek } from '../utils/constants';
+import {
+  StyleSheet,
+  View,
+  ViewStyle,
+  Text,
+  TextStyle,
+  TextInput,
+  TouchableOpacity
+} from 'react-native';
+
+interface Styles {
+  title: TextStyle;
+  container: ViewStyle;
+  label: TextStyle;
+  input: TextStyle;
+  inputWrapper: ViewStyle;
+  button: ViewStyle;
+  daysOfTheWeekContainer: ViewStyle;
+  selectedDaysOfWeek: ViewStyle;
+}
+
+interface IValues {
+  name?: string;
+  description?: string;
+  motivation?: string;
+  reminder?: string;
+  selectedDays?: any;
+  date?: any;
+}
+interface IProps {
+  navigation?: {
+    navigate: Function;
+  };
+}
+
+export default function ActivityPage(props: IProps) {
+  const [values, setValues] = useState<IValues>({ date: new Date() });
+  const [selectedDays, setSelectedDays] = useState<string[]>([]);
+
+  function handleChange(name: string, event: string): void {
+    setValues({
+      ...values,
+      [name]: event
+    });
+  }
+
+  function handleDayOfWeekClick(day: string) {
+    if (selectedDays.includes(day)) {
+      const unselect = selectedDays.filter(selectedDay => selectedDay != day);
+      setSelectedDays(unselect);
+      setValues({
+        ...values,
+        selectedDays: selectedDays
+      });
+    } else {
+      setSelectedDays([...selectedDays, day]);
+      setValues({
+        ...values,
+        selectedDays: selectedDays
+      });
+    }
+  }
 
-export default function ActivityPage() {
   return (
-    <View>
-      <Text>This is the ActivityPage</Text>
+    <View style={styles.container}>
+      <Text style={styles.title}>Setup Activity</Text>
+      <View>
+        <View style={styles.inputWrapper}>
+          <Text style={styles.label} testID="activity-name">
+            Activity Name? (required)
+          </Text>
+          <TextInput
```

Comment:
is there a way to mark this as required.  or how are we making sure that this and the other required fields are actually required?

Reply:
okay, makes sense 👏🏽